planning/behavior_velocity_planner/autoware_behavior_velocity_blind_spot_module/src/util.cpp

```diff
@@ -0,0 +1,314 @@
+{
+  for (const auto & prev : routing_graph_ptr->previous(assigned_lane)) {
+    for (const auto & following : routing_graph_ptr->following(prev)) {
+      if (std::string(following.attributeOr("turn_direction", "else")) == "straight") {
+        return following;
+      }
+    }
+  }
+  return std::nullopt;
+}
+
+lanelet::ConstLanelet generateHalfLanelet(
+  const lanelet::ConstLanelet lanelet, const TurnDirection & turn_direction,
+  const double ignore_width_from_centerline)
+{
+  lanelet::Points3d lefts, rights;
+
+  const double offset = (turn_direction == TurnDirection::LEFT) ? ignore_width_from_centerline
+                                                                : -ignore_width_from_centerline;
+  const auto offset_centerline = lanelet::utils::getCenterlineWithOffset(lanelet, offset);
+
+  const auto original_left_bound =
+    (turn_direction == TurnDirection::LEFT) ? lanelet.leftBound() : offset_centerline;
+  const auto original_right_bound =
+    (turn_direction == TurnDirection::LEFT) ? offset_centerline : lanelet.rightBound();
+
+  for (const auto & pt : original_left_bound) {
+    lefts.emplace_back(pt);
+  }
+  for (const auto & pt : original_right_bound) {
+    rights.emplace_back(pt);
+  }
+  const auto left_bound = lanelet::LineString3d(lanelet::InvalId, std::move(lefts));
+  const auto right_bound = lanelet::LineString3d(lanelet::InvalId, std::move(rights));
+  auto half_lanelet = lanelet::Lanelet(lanelet::InvalId, left_bound, right_bound);
+  return half_lanelet;
+}
+
+lanelet::ConstLanelet generateExtendedAdjacentLanelet(
+  const lanelet::ConstLanelet lanelet, const TurnDirection direction,
+  const double adjacent_extend_width)
+{
+  const auto centerline = lanelet.centerline2d();
+  const auto width =
+    boost::geometry::area(lanelet.polygon2d().basicPolygon()) / boost::geometry::length(centerline);
+  const double extend_width = std::min<double>(adjacent_extend_width, width);
+  const auto left_bound_ =
+    direction == TurnDirection::LEFT
+      ? lanelet::utils::getCenterlineWithOffset(lanelet, -width / 2 + extend_width)
+      : lanelet.leftBound();
+  const auto right_bound_ =
+    direction == TurnDirection::RIGHT
+      ? lanelet::utils::getCenterlineWithOffset(lanelet, width / 2 - extend_width)
+      : lanelet.rightBound();
+  lanelet::Points3d lefts, rights;
+  for (const auto & pt : left_bound_) {
+    lefts.emplace_back(pt);
+  }
+  for (const auto & pt : right_bound_) {
+    rights.emplace_back(pt);
+  }
+  const auto left_bound = lanelet::LineString3d(lanelet::InvalId, std::move(lefts));
+  const auto right_bound = lanelet::LineString3d(lanelet::InvalId, std::move(rights));
+  auto new_lanelet = lanelet::Lanelet(lanelet::InvalId, left_bound, right_bound);
+  const auto new_centerline = lanelet::utils::generateFineCenterline(new_lanelet, 5.0);
+  new_lanelet.setCenterline(new_centerline);
+  return new_lanelet;
+}
+
+lanelet::ConstLanelet generateExtendedOppositeAdjacentLanelet(
```

Comment:
- [X] same impl as before
- [x] check if `planner_param_.opposite_adjacent_extend_width` is passed as before